runtime/metricsview/astexpr.go

```diff
@@ -316,6 +339,10 @@ func (b *sqlExprBuilder) writeBinaryConditionInner(left, right *Expression, left
 
 	b.writeByte(')')
 
+	if leftOverrideNeedsClosure {
+		b.writeByte(')')
+	}
```

Comment:
Can this be added around line 229 instead of being pushed into these functions? Or can you think of a refactor that would enable the open and close parenthesis to be written together?

It's just that adding the closing parenthesis in another scope than the open parenthesis makes it very difficult to understand if the parentheses are closed correctly or not in all cases.